ClosedXML_Tests/Excel/CalcEngine/FunctionsTests.cs

```diff
@@ -658,5 +659,131 @@ public void TestOmittedParameters()
                 Assert.AreEqual(2, value);
             }
         }
+
+        [Test]
+        public void NumberValue()
+        {
+            Object actual;
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.45\", \".\", \",\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123,45\", \",\", \".\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.456.789,01\", \",\", \".\")");
+            Assert.AreEqual(123456789.01, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,234.56\", \".\", \",\")");
+            Assert.AreEqual(1234.56, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1.234,56\", \",\", \".\")");
+            Assert.AreEqual(1234.56, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.45\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,234.56\")");
+            Assert.AreEqual(1234.56, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"+1\")");
+            Assert.AreEqual(1, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"+1.23\")");
+            Assert.AreEqual(1.23, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"-1\")");
+            Assert.AreEqual(-1, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"-1.23\")");
+            Assert.AreEqual(-1.23, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\".1\")");
+            Assert.AreEqual(0.1, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1.234567890E+34\")");
+            Assert.AreEqual(1.234567890E+34, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\".99999999999999\")");
+            Assert.AreEqual(0.99999999999999, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"-.99999999999999\")");
+            Assert.AreEqual(-0.99999999999999, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,23,4\")");
+            Assert.AreEqual(1234, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,234,56\")");
+            Assert.AreEqual(123456, actual);
+        }
+
+        [Test]
+        public void NumberValueExceptions()
+        {
+            Object actual;
+
+            try
```

Comment:
Use `Assert.Throws<CellValueException>` to test the exceptions. Don't try/catch. And definitely don't check the message... that could easily change in the future.